ffc/process_charge_file.py

```diff
@@ -0,0 +1,64 @@
+import asyncio
+import logging
+import tempfile
+from typing import Any
+
+from ffc.client import get_httpx_ffc_api_client
+
+logger = logging.getLogger(__name__)
+
+
+class ChargeFileSplitProcessor:
+    def __init__(self, mtp_base_api_url: str):
+        self.finops_client = get_httpx_ffc_api_client()
+        self.MPT_API_BASE_URL = mtp_base_api_url
+
+    async def process_generated_charge_files(self) -> list[Any] | None:
+        """
+        This method downloads all the charge files in the GENERATED status
+        and for each of them it will download the related zip file from Azure Blob Storage
+        The charge file in the GENERATED status is a list like the following example
+            {
+                "items": [
+                    {
+                        "amount": 100.4,
+                        "currency": "USD",
+                        "document_date": "2017-05-19T00:00:00",
+                        "events": {
+                            "created": {"at": "2025-04-28T10:32:16.100842Z"},
+                            "updated": {"at": "2025-04-28T10:32:16.100848Z"},
+                        },
+                        "id": "FCHG-5825-2145-4566",
+                        "owner": {"id": "FACC-4174-5917", "name": "Test", "type": "operations"},
+                        "status": "generated",
+                    }
+                ],
+                "limit": 50,
+                "offset": 0,
+                "total": 1,
+            }
+        Returns : A list of the paths where the charge files are stored, like
+        ['/var/folders/07/7_4bnbgx3xg8845xyq266g1w0000gn/T/FCHG-1035-3738-7350.zip']
+        """
+        try:
+            temp_dir = tempfile.gettempdir()
+            charge_file_list = await self.finops_client.get_generated_charge_files()
+            tasks = [
+                self.finops_client.get_charges_file_download_url(
```

Comment:
As we discussed, we don't want to parallelize only the downloads but we want an end to end processing of  X charges file at once. So I would suggest either have a method of this splitter called `def process_charges_file(charges_file_id: str)` that, for the moment, it only runs the download and in this method you create a task that run such method for each `charges_file_id`, either remove the method completely.